lib/lib_rss.php

```diff
@@ -48,7 +48,31 @@ function classAutoloader($class) {
 	} elseif (strpos($class, 'CssXPath') !== false) {
 		include(LIB_PATH . '/CssXPath/' . basename(str_replace('\\', '/', $class)) . '.php');
 	} elseif (strpos($class, 'PHPMailer') === 0) {
-		include(LIB_PATH . '/' . str_replace('\\', '/', $class) . '.php');
+		// project-specific namespace prefix
+		$prefix = 'PHPMailer\\';
+
+		// base directory for the namespace prefix
+		$base_dir = LIB_PATH . '/phpmailer/phpmailer/src/';
+
+		// does the class use the namespace prefix?
+		$len = strlen($prefix);
+		if (strncmp($prefix, $class, $len) !== 0) {
+			// no, move to the next registered autoloader
+			return;
+		}
+
+		// get the relative class name
+		$relative_class = substr($class, $len);
+
+		// replace the namespace prefix with the base directory, replace namespace
+		// separators with directory separators in the relative class name, append
+		// with .php
+		$file = $base_dir . str_replace('\\', '/', $relative_class) . '.php';
+
+		// if the file exists, require it
+		if (file_exists($file)) {
+			require $file;
+		}
```

Comment:
Any problem with the previous code?

Reply:
Composer installs the dependencies inside the folder structure `<vendorname_lowercase>/<packagename_lowercase>`. The `src` folder is defined by the PHPMailer package. So the previous code wont work and I didn't want to bother with more details for the autoloader for this proposal. So just copied the general example code for a PSR-4 implementation. As you correctly noted, this could be further improved.

https://www.php-fig.org/psr/psr-4/examples/#closure-example